infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -506,7 +525,15 @@ public void prepareSiteRoleChange() {
      */
     @Override
     public int queryReplicationStatus() {
-        return replicationManager.queryReplicationStatus();
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
```

Comment:
Add log.warn if the roletype is not active.

Reply:
Yeah, I think I need some clarifications for this function.

Case 1: if the role is standby, what is the expected return value?
Case 2: if the role is active, but somehow replication is not initialized, like have not got the lock. What is the expected value?

Will add log.warn

Reply:
Case 1, this should not be called on the standby side. As standby side doesn't know the active log tail information. 
But for snapshot sync scenario during applying phase, the active can query standby querySnapshotProcessStatus(), the standby side reply how much data has been appied. This can be done at phase II implementation.
Case 2, do you mean the active is not the leader node? For now, we can consider it should be called on  the leader node only. We need to talk to the other group and get their finalized API and implement the complete solution.

Reply:
Yeah, I understand both cases should not happen right now, and just to avoid NPE. I have added warning logs for both cases.